tools/scalar-schema/src/scalar_schema/jdbc.clj

```diff
@@ -0,0 +1,294 @@
+       (->> (map enclosure-fn ["full_table_name" "column_name"])
+            (string/join \,))
+       "))"))
+
+(defn- create-metadata-table
+  [{:keys [rdb-engine execution-fn] :as opts}]
+  (try
+    (execution-fn (make-create-metadata-statement opts))
+    (log/info "metatable table is created successfully.")
+    (catch SQLException e (if (string/includes? (.getMessage e) "already") nil (throw e)))))
+
+(defn- make-create-database-statement
+  [database {:keys [enclosure-fn]}]
+  (str "CREATE SCHEMA " (enclosure-fn database)))
+
+(defn- create-database
+  [database {:keys [rdb-engine execution-fn] :as opts}]
+  (cond
+    (= rdb-engine :oracle) nil
+    :else (try
+            (execution-fn (make-create-database-statement database opts))
+            (log/info "The database" database "is created successfully.")
+            (catch SQLException e (if (or (string/includes? (.getMessage e) "already")
+                                          (string/includes? (.getMessage e) "database exists"))
+                                    nil (throw e))))))
+
+(defn- create-metadata-database
+  [{:keys [prefix] :as opts}]
+  (let [meta-database common/METADATA_DATABASE
+        database (if prefix (str prefix \_ meta-database) meta-database)]
+    (create-database database opts)))
+
+(defn- make-create-table-statement
+  [{:keys [database table partition-key clustering-key reordered-columns secondary-index]}
+   {:keys [enclosure-fn data-type-fn] :as opts}]
+  (let [primary-key (concat partition-key clustering-key)]
+    (str "CREATE TABLE "
+         (get-table-name database table opts)
+         \(
+         (->> (map #(let [column (enclosure-fn (first %))
+                          data-type (data-type-fn (first %) (second %))]
+                      (str column \  data-type)) reordered-columns)
+              (string/join \,))
+         ", PRIMARY KEY ("
+         (->> (map enclosure-fn primary-key)
+              (string/join \,))
+         "))")))
+
+(defn- create-table
+  [schema {:keys [execution-fn] :as opts}]
+  (execution-fn (make-create-table-statement schema opts)))
+
+(defn- make-create-index-statement
+  [{:keys [database table]} {:keys [enclosure-fn] :as opts} indexed-column]
+  (let [index-name (str common/INDEX_NAME_PREFIX \_ database \_ table \_ indexed-column)]
+    (str "CREATE INDEX "
+         (enclosure-fn index-name)
+         " ON "
+         (get-table-name database table opts)
+         \(
+         (enclosure-fn indexed-column)
+         \))))
+
+(defn- create-index
+  [{:keys [secondary-index] :as schema} {:keys [execution-fn] :as opts}]
+  (doall (map #(execution-fn (make-create-index-statement schema opts %)) secondary-index)))
+
+(defn- get-key-type
+  [column partition-key clustering-key]
+  (cond
+    (key? column partition-key) "PARTITION"
+    (key? column clustering-key) "CLUSTERING"
+    :else nil))
+
+(defn- is-indexed [column secondary-index]
+  (if (key? column secondary-index) true false))
```

Comment:
If the above change would be applied, it could evaluate just the `(key? ...)` expression.